src/data/taskList.ts

```diff
@@ -0,0 +1,72 @@
+import { SystemIntakeForm } from 'types/systemIntake';
+import { BusinessCaseModel } from 'types/businessCase';
+import { businessCaseInitialData } from 'data/businessCase';
+
+export const calculateIntakeStatus = (intake: SystemIntakeForm) => {
+  if (intake.id === '') {
+    return 'START';
+  }
+  if (intake.status === 'DRAFT') {
+    return 'CONTINUE';
+  }
+  return 'COMPLETED';
+};
+
+export const chooseIntakeLink = (intake: SystemIntakeForm, status: string) => {
+  const newIntakeLink = '/system/new';
+  if (intake.id === '') {
+    return newIntakeLink;
+  }
+  let link: string;
+  switch (status) {
+    case 'CONTINUE':
+      link = `/system/${intake.id}/contact-details`;
+      break;
+    case 'COMPLETED':
+      // This will need to be changed once we have an intake review page
+      link = '/';
+      break;
+    default:
+      link = newIntakeLink;
+  }
+  return link;
+};
+
+export const calculateIntakeFeedbackStatus = (intakeStatus: string) => {
+  switch (intakeStatus) {
+    case 'SUBMITTED':
```

Comment:
If you're just mapping string here, then you could potentially do this with an object rather than a function. eg:
```
export const intakeFeedbackStatus = (intakeStatus: string) => (
  {
    SUBMITTED: 'SUBMITTED',
    CLOSED: 'COMPLETED'
  }[intakeStatus]
)
```

You might be able to use this strategy for some of the other functions in this file, too. Also, I dropped `calculate` from the function name. I think it's unnecessary, but I'm ambivalent if you want to keep it.

Reply:
I named it calculate to differentiate the value we're using in the form.

Reply:
This reads more easily because the switch case shows the groupings of the intake statuses more clearly for debugging and handles the default case more explicitly. Plus this doesn't require any memory allocation.

Reply:
For me, the object is much easier to read than the switch statement. But I hear you on the default value being more explicit. I'm unsure of the memory allocation here, but that's not something I would worry about too much at this point, unless we're discussing dropping LARGE data sets into a reducer.